Frontegg OIDC and SAML Integration Guide (#414)
ngrok/ngrok-docs: docs/integrations/frontegg/sso-oidc.md

---

> description: Use Frontegg OIDC to secure access to ngrok tunnels

Reviewer: ```suggestion
description: Use Frontegg OIDC to secure access to ngrok endpoints
```

---

# Frontegg SSO (OpenID Connect)

---

:::tip TL;DR

> To secure access to ngrok with Frontegg Single Sign-On using OpenID Connect:

Reviewer: ```suggestion
To secure access to ngrok with Frontegg Single Sign-On (SSO) using OpenID Connect:
```

1. [Configure Frontegg SSO](#configure-frontegg)
1. [Configure ngrok](#configure-ngrok)
1. [Test access to ngrok with Frontegg SSO](#test-sso)

:::

> This article details how to configure Frontegg as the primary Identity Provider for ngrok tunnels.

Reviewer: ```suggestion
This article details how to configure Frontegg as the primary Identity Provider (IDP) for ngrok endpoints.
```

By integrating Frontegg SSO with ngrok, you can:

> - **Restrict access to ngrok tunnels** only to users authenticated via Frontegg

Reviewer: ```suggestion
- **Restrict access to ngrok endpoints** only to users authenticated via Frontegg
```

- **Use Frontegg security policies and MFA authenticators**.
- **Use Frontegg's Dashboard to facilitate access to ngrok apps**.

## Supported Features

The ngrok integration with Frontegg supports:

> - **SP-Initiated SSO**: In this mode, users access ngrok edges and tunnels and are redirected to Frontegg for authentication.

Reviewer: ```suggestion
- **SP-Initiated SSO**: In this mode, users access ngrok endpoints and are redirected to Frontegg for authentication.
```

## Requirements

> To configure ngrok tunnels with Frontegg, you must have:

Reviewer: ```suggestion
To configure an ngrok edge with Frontegg, you must have:
```

> - a Frontegg account with administrative rights to create apps

Reviewer: ```suggestion
- A Frontegg account with administrative rights to create apps
```

> - an [ngrok Enterprise Account](https://ngrok.com/pricing) with an authtoken or admin access to configure edges with OpenID Connect.

Reviewer: ```suggestion
- An [ngrok Enterprise Account](https://ngrok.com/pricing) with an authtoken or admin access to configure edges with OpenID Connect.
```

1. Click the **Domains** tab, and copy the value of the **Domain name**.

### Grant access to Frontegg users

Create a Frontegg user:

1. On the left menu of the [Frontegg Portal](https://portal.frontegg.com/), click your environment, click **Backoffice**, click **Accounts**, and then click **Add New**.

1. On the **Add new account** popup, enter a name for the account, `ngrok.com` in the **Domain** field, and then click **Save**.

1. On the left menu, click **Backoffice**, click **Users**, and then click **Add New**.

1. On the **Add new user** popup, enter a name, email, select the account you have created, and then click **Save**.
   **Note**: By default, the user is assigned the **Admin** role.

## **Step 2**: Configure ngrok {#configure-ngrok}

ngrok can leverage Frontegg SSO in two ways:

- From the ngrok CLI (using the `--oidc` parameter)
- From the ngrok dashboard

### **Option 1**: ngrok CLI

> **Note:** For this tutorial, we assume you have an app running locally (i.e., on localhost:3000) with the ngrok client installed.

1. Launch a terminal

1. Enter the following command to launch an ngrok tunnel with Frontegg SSO:

   ```bash

>    ngrok http 3000 --oidc=Frontegg_OAUTH_URL \

Reviewer: the standard for env variables IS_ALL_CAPS. can we change this to FRONTEGG_OAUTH_URL? same with the client id and secret.

Author: You're right. Fixed.
